docs/examples/patterns/contextual-menu/_script.js

```diff
@@ -0,0 +1,78 @@
+/**
+  Toggles the necessary aria- attributes' values on the menus
+  and handles to show or hide them.
+  @param {HTMLElement} element The menu link or button.
+  @param {Boolean} show Whether to show or hide the menu.
+  @param {Number} top Top offset in pixels where to show the menu.
+*/
+function toggleMenu(element, show, top) {
+  var target = document.getElementById(element.getAttribute('aria-controls'));
+
+  if (target) {
+    element.setAttribute('aria-expanded', show);
+    target.setAttribute('aria-hidden', !show);
+
+    if (typeof top !== 'undefined') {
+      target.style.top = top + 'px';
```

Comment:
My worry with this is that this will be the first time we apply styling via JS not just functionality. I think if people use the mark up in the example they should get what they see in the example. I know that the JS needs to be used but simply toggling aria is easier to get right and document.

Reply:
@anthonydillon 

Ok. Stepping back a bit.

We currently have an issue that contextual menu is misaligned (too much to the top):

<img width="1031" alt="Screenshot 2020-02-13 at 12 43 05" src="https://user-images.githubusercontent.com/83575/74433202-40d8d300-4e60-11ea-84e5-80b38e4ed3f7.png">

It's because we have some styling that adds negative margin (possibly to compensate for negative margin on button).
If we remove any compensation and changes to margins we get:

<img width="1029" alt="Screenshot 2020-02-13 at 12 55 45" src="https://user-images.githubusercontent.com/83575/74433276-61a12880-4e60-11ea-97e1-746b3d23453e.png">

Which fixes issue for inline links, but makes it wrong the opposite way for buttons.
So it's hard to call it 'fix'. The margin bottom on button is large enough that contextual menu looks visually unattached. 

If we can leave with that, we can leave it like this. And document that you could add `u-no-margin` to this button or fix it with JS.

Or, we could fix it with JS ourselves as well - as an example.
Yes, it's more complicated that toggling visibility. But there is no good way around it. It either looks bad and broken or we have more code somewhere (weird custom complex rules in SCSS or a bit of positioning in JS). If we want Vanilla CSS to be cleaner (because we can't simply cover all possible cases for stuff with different margins) we need to have a bit more complex JS. And for contextual menu you need some custom JS anyway. It's likely that its main use is in apps rather than websites, so they already have custom JS to handle showing this dropdown and all the actions it triggers.

On the opposite side - for Vanilla tooltips, which are currently CSS only - there were number of complaints form apps devs that they are impossible to control and position with JS (which is how apps handle them). So I think it's OK to leave some of responsibility of positioning to JS. 

Both for tooltips and dropdowns you may easily get to the point where you want to change their position (top/bottom/left/right) depending on where they are on screen, so they are not off screen when opened. We can't do it from Vanilla CSS. It would only be possible with JS.

Bootstrap doesn't even bother - they use external library for dropdowns and position them with position absolute. They control styling, but position is controlled by JS. 

It's not perfect, but it's hard to avoid if we want to avoid having exceptions in SCSS.

Reply:
I agree, fair point :+1: